### ISSUE TITLE
TensorFlow backend 3-4 times slower than Theano ? 

### ISSUE THREAD

**Issue by david-droz** [keras-team/keras#10206]
Versions:
* Keras : 2.1.6
* Tensorflow : 1.7.0
* Theano : 1.0.1
* CUDA / cuDNN : 9.1.85 / 7.0.5
* libgpuarray : 0.7.5 

Hardware (GPU): Tesla P100-PCIE-12GB

Model: 4 dense layers with some dropout and Adam optimiser. 
```
model = Sequential()
model.add(Dense(200,input_shape=(47,),kernel_initializer='he_uniform',activation='relu'))
model.add(Dropout(0.3))
model.add(Dense(100,kernel_initializer='he_uniform',activation='relu'))
model.add(Dropout(0.3))
model.add(Dense(50,kernel_initializer='he_uniform',activation='relu'))
model.add(Dropout(0.2))
model.add(Dense(1,kernel_initializer='he_uniform',activation='sigmoid'))
model.compile(loss='binary_crossentropy', optimizer=Adam(), metrics=['binary_accuracy'])
```

Full code: https://gist.github.com/david-droz/d0edfc4ebd3fda09c13521ff515d8529#file-keras_theano_tf_issue-py-L89

Average train time per epoch (100 epochs - 200k samples):
* Theano : 8.90 seconds
* TensorFlow : 31.82 seconds

Loss at last epoch (train/test):
* Theano : 0.0228 / 0.0196
* TensorFlow : 0.0223 / 0.0195

Binary accuracy at last epoch (train/test):
* Theano : 0.9933 / 0.9939
* TensorFlow : 0.9936 / 0.9944

-------------------

It seems that TensorFlow is significantly slower, even on such a relatively minimal model and on low-dimensional data (47 variables per observation). While there is some gain in accuracy, similar gain can be attained by simply training Theano longer. Why is that?

**Comment by JamesTrick** [comment]
Hey david-droz

Such a detailed post! I can't fully comment on the specifics but #2156 was raised in 2016 which talks about the speed differences. It's believed that Theano is consistently faster than Tensorflow, at times up to 4x faster.

There might be some good steps in there to help you out?

Cheers
James
